### PR TITLE
[montecarlo] Fix enum PDG_t

### PR DESCRIPTION
**Pull request by hahnjo**
`LinkDef.h` specifies a `pragma link C++ enum PDG_t` but actually `PDG_t` was a `typedef` to an anonymous `enum`.

### CODE REVIEW

**Comment by hahnjo** [comment]
As far as I can tell, this is the only occurrence of `pragma link enum` to a `typedef`. I found this because `rootcling` complained with the attempt to upgrade to LLVM 16 while it worked (maybe by chance) before. Do you think it would be ok to break this corner case? @pcanal @vgvassilev

**Comment by vgvassilev** [comment]
This is an idiom widely used in C. I suspect that will have quite a big impact to people's dictionaries and in particular when they request dictionaries of idiomatic C codes...

**Comment by hahnjo** [comment]
> This is an idiom widely used in C. I suspect that will have quite a big impact to people's dictionaries and in particular when they request dictionaries of idiomatic C codes...

Ha, it's actually the other way around and it fails with LLVM 16 because that one makes a bigger effort to "see through" `typedef`s and finally finds `PDG_t`. The reason we don't notice so far is that non-existing `enums` referenced in `LinkDef.h` files don't trigger any error / warning - just try `#pragma link C++ enum DOES_NOT_EXIST;`. @pcanal do you want an issue for that?

tldr: this change actually *fixes* `PDG_t` to be included. Alternatively, we can also just remove it entirely as it's never been found so far...

**Comment by hahnjo** [comment]
ping; as explained above, this is actually improving the situation. `typedef`s to anonymous `enum`s apparently never worked...

**Comment by Axel-Naumann** [comment]
I think we need to discuss:

- what the effect of a LinkDef entry of an enum is today, and what it is supposed to be
- whether we should warn / error out for LinkDef entries for non-existing enums
- whether LinkDef entries through typedef should work, in general

Maybe we use today's IO meeting for that, but it'd be good to have some of the answers ready by that time.

**Comment by Axel-Naumann** [comment]
Agreed today at I/O meeting that we need to remind ourselves and investigate these questions. I'm okay with merging this PR if e.g. @hahnjo could create a new issue with the above questions - this seems to indicate a bug, whether it's a missing error message or an ignored linkdef entry or whatever it is.

**Review by Axel-Naumann — APPROVED** [review]

**Comment by hahnjo** [comment]
> Agreed today at I/O meeting that we need to remind ourselves and investigate these questions. I'm okay with merging this PR if e.g. @hahnjo could create a new issue with the above questions - this seems to indicate a bug, whether it's a missing error message or an ignored linkdef entry or whatever it is.

https://github.com/root-project/root/issues/13303
